public/router.php

```diff
@@ -25,15 +25,21 @@
 	date_default_timezone_set('UTC');
 }
 
-define('BASE_DIR', dirname(__DIR__).'/');
+$baseDir = str_replace('\\', '/', dirname(dirname(__FILE__)).'/');
+if (!is_file($baseDir.'pufferpanel')) {
+    $baseUrl = dirname($_SERVER['PHP_SELF']);
```

Comment:
Master url will always exist and be filled in.
This is not the best nor easiest solution. 

We will deal with making master url required. All that really matters there is the subfolder, which you can assume it's / if there's no master url.